src/public/modules/forms/services/submissions.client.factory.js

```diff
@@ -93,18 +93,20 @@ function SubmissionsFactory(
     xhr.send(fd)
     // On Response
     xhr.onreadystatechange = function () {
-      // 4 DONE  The operation is complete.
       const OPERATION_DONE = 4
       if (xhr.readyState === OPERATION_DONE) {
         // waterfall is successful
+        let response = {}
+        try {
+          response = JSON.parse(xhr.responseText)
+          // eslint-disable-next-line no-empty
+        } catch (e) {}
```

Comment:
we shouldn't ever leave errors unhandled (as the eslint rule suggests haha) - if the response cannot be parsed, we should do `deferred.reject`